CORPORATE_SETUP.md

```diff
@@ -0,0 +1,111 @@
+
+# Setup Guide for Mac Users with Corporate git credentials.
+
+This guide is for corporate contributors who are already using corporate git accounts and need to configure their system to support using a separate public github account in addition to their pre-existing corporate account.
+
+
+## Setup SSH config for multiple hosts.
+
+### Step 1: Setup a github.com user account if you don't already have one.
+
+Use a personal email account, not your corporate email account.
+
+### Step 2: Generate a new ssh key for your github.com account
+
+Follow these steps with these notes:
+
+1. Use the same personal email address that you used in step 1.
+2. Do not use the default id_rsa name, since that's probably already in use for your corporate ssh key. Use "id_rsa_github".
+
+https://help.github.com/en/github/authenticating-to-github/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent
+
+### Step 3: Edit your ~/.ssh/config file to use different ssh keys
+
+This is mine for reference. Make sure you DO NOT have an IdentifyFile setting under the Host * section.
+
+    Host *
+        AddKeysToAgent yes
+        UseKeychain yes
+
+    Host git.corp.adobe.com
+        IdentityFile ~/.ssh/id_rsa
+        User git
+        IdentitiesOnly yes
+
+    Host github.com
+        IdentityFile ~/.ssh/id_rsa_github
+        User git
+        IdentitiesOnly yes
+
+## Setup GIT config for multiple hosts.
+
+### Step 4: Create a separate root directory for public github.com projects
+
+I use `~/Projects/public`
+
+### Step 5: Create a .gitconfig in that directory setting the user to your public email address
+
+`~/Projects/public/.gitconfig`
+
+    [user]
+        email = youremail@gmail.com
+        name = Your Name
+
+### Step 6: Add an [includeIf] section to your ~/.gitconfig telling it to use the new .gitconfig within that directory
+
+`~/.gitconfig`
+
+    [user]
+        name = Your Name
+        email = youremail@yourcorporation.com
+    [merge]
+        tool = p4mergetool
+    [mergetool "p4mergetool"]
```

Comment:
maybe trim this section down, not everyone has access to p4merge outside of adobe

Reply:
can do